sympy/parsing/C/parser.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import unicode_literals, print_function
+from sympy.parsing.sympy_parser import (parse_expr, standard_transformations, implicit_multiplication_application)
+from clang.cindex import (
+    CursorKind,
+    Index,
+    StorageClass,
+    TranslationUnit,
+    TypeKind,
+    UnaryOperator
+)
+
+import argparse
+import os
+import re
+import sys
+
+
+class Writer(object):
```

Comment:
Why no inheriting `Basic`? Just asking out of curiosity.

Reply:
No particular reason.
I can inherit from basic if suggest so?

Reply:
I think you should do it.

Reply:
What would be the advantage of inheriting from Basic? I don't think this class is related to it at all.

Reply:
I am suggesting for consistency.  If it's not required then no issues.  We can go without it.

Reply:
I would only inherit from Basic if the class is supposed to be part of a SymPy expression, which this one does not.

Reply:
Please resolve this conversation. Thanks @certik for the reason.